lib/simulator-xcode-9.js

```diff
@@ -160,6 +149,34 @@ class SimulatorXcode9 extends SimulatorXcode8 {
     }
   }
 
+  /***
+   * Boots simulator and opens simulators UI Client if not already opened.
+   *
+   * @param uiClientPid - process id of simulator UI client.
+   * @param opts - arguments to start simulator UI client with.
+   * @returns {Promise} - a promise that resolves when sim is booted and sim UI client is launched.
+   */
+  async launchSimWindow (uiClientPid, opts) {
+    log.info(`Booting Simulator with UDID '${this.udid}'...`);
+    await this.bootSimulator();
+    if (!uiClientPid) {
+      await this.startUIClient(opts);
+    }
+  }
+
+  /**
+   * Boots simulator.
+   *
+   * @returns {Promise} - A promise that resolved when simulator is booted.
+   */
+  async bootSimulator () {
+    try {
+      await retryInterval(3, 2000, async () => await this.simctl.bootDevice());
```

Comment:
this call might cause unnecessary delays if the simulator is already booted. The original command being executed more than once returns non-zero exit code:

```
An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=165):
Unable to boot device in current state: Booted
```

Reply:
Ok that's a good point, I scanned [run(opts = {})](https://github.com/sadafyadegari/appium-ios-simulator/blob/master/lib/simulator-xcode-9.js#L67) and it shouldn't call BootSimulator when `serverState === 'Booted'` . But I agree that it'd be good to add a guard against that case for posterity. I'll add a sim state check inside `bootSimulator` function.